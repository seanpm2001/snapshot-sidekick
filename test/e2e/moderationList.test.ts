import request from 'supertest';

const HOST = `http://localhost:${process.env.PORT || 3000}`;

describe('GET /moderationList', () => {
  describe('when fields params is empty', () => {
    it('returns all the list', async () => {
      const response = await request(HOST).get('/moderationList');

      expect(response.statusCode).toBe(200);
      expect(response.body).toMatchSnapshot();
    });
  });

  describe('when fields params is set', () => {
    it.each(['flaggedLinks', 'verifiedSpaces', 'flaggedProposalIds'])(
      'returns only the selected %s list',
      async field => {
        const response = await request(HOST).get(`/moderationList?fields=${field}`);

        expect(response.statusCode).toBe(200);
        expect(response.body).toMatchSnapshot();
      }
    );
  });

  it('returns multiple list: verifiedSpaces,flaggedProposalIds', async () => {
    const response = await request(HOST).get(
      `/moderationList?fields=verifiedSpaces,flaggedProposalIds`
    );

    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('ignores invalid field, and returns only verifiedSpaces', async () => {
    const response = await request(HOST).get(`/moderationList?fields=verifiedSpaces,testInvalid`);

    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
