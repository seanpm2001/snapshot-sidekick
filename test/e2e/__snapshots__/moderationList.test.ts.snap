// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /moderationList ignores invalid field, and returns only verifiedSpaces 1`] = `
{
  "verifiedSpaces": {
    "space1.eth": 1,
    "space2.eth": 2,
  },
}
`;

exports[`GET /moderationList returns multiple list: verifiedSpaces,flaggedProposals 1`] = `
{
  "flaggedProposals": [
    "0x1",
    "0x2",
  ],
  "verifiedSpaces": {
    "space1.eth": 1,
    "space2.eth": 2,
  },
}
`;

exports[`GET /moderationList when fields params is empty returns all the list 1`] = `
{
  "flaggedLinks": [
    "https://gogle.com",
    "https://facebook.com",
  ],
  "flaggedProposals": [
    "0x1",
    "0x2",
  ],
  "verifiedSpaces": {
    "space1.eth": 1,
    "space2.eth": 2,
  },
}
`;

exports[`GET /moderationList when fields params is set returns only the selected flaggedLinks list 1`] = `
{
  "flaggedLinks": [
    "https://gogle.com",
    "https://facebook.com",
  ],
}
`;

exports[`GET /moderationList when fields params is set returns only the selected flaggedProposals list 1`] = `
{
  "flaggedProposals": [
    "0x1",
    "0x2",
  ],
}
`;

exports[`GET /moderationList when fields params is set returns only the selected verifiedSpaces list 1`] = `
{
  "verifiedSpaces": {
    "space1.eth": 1,
    "space2.eth": 2,
  },
}
`;
